Add data type chips to summary and detail sections in evidence page

mirandaio opened the pull request:
Closes https://github.com/opentargets/platform/issues/1493

mirandaio commented on src/sections/evidence/EuropePmc/Body.js:
```diff
@@ -59,29 +59,32 @@ const columns = [
-  const mergedRows = [...rows];
-
-  rows.forEach(row => {
+function mergeData(rows, literatureData) {
```
This is a fix to a bug unrelated to the PR but it was blocking my work because it was throwing an error in dev mode.

LucaFumis reviewed (approved):
Thanks @mirandaio - looks great. I would have perhaps added a config to map datasources and datatypes instead of hard-coding the labels for each source, even though they won't change frequently (so not necessarily error prone). This can always be refactored at later stage.

mirandaio commented:
Thanks for the suggestion @LucaFumis I've addressed your comment by reusing the data types from https://github.com/opentargets/platform-app/blob/main/src/dataTypes.js
